src/CommunityToolkit.Maui.Core/IPopupService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.ComponentModel;
+
+namespace CommunityToolkit.Maui.Core;
+
+/// <summary>
+/// Provides a mechanism for displaying <see cref="CommunityToolkit.Maui.Core.IPopup"/>s based on the underlying view model.
+/// </summary>
+public interface IPopupService
+{
+	/// <summary>
+	/// Resolves and displays a <see cref="CommunityToolkit.Maui.Core.IPopup"/> and <typeparamref name="TViewModel"/> pair that was registered with <c>AddTransientPopup</c>.
+	/// </summary>
+	/// <typeparam name="TViewModel">The type of the view model registered with the <see cref="CommunityToolkit.Maui.Core.IPopup"/>.</typeparam>
+	void ShowPopup<TViewModel>() where TViewModel : INotifyPropertyChanged;
```

Comment:
Is INotifyPropertyChanged constraint really necessary here? Can we use class instead?

Reply:
The PopupService is created to facilitate using Popups with the MVVM architecture.

In the MVVM architecture, the way the ViewModel communicates updates to the View is via `INotifyPropertyChanged`. If your class isn't implementing `INotifyPropertyChanged`, then it isn't really a ViewModel and doesn't require this method; you can use the other, existing, methods to display your Popup: https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/views/popup#presenting-a-popup

Reply:
ViewModel may have only commands and in this case INotifyPropertyChanged is not needed, I just don't like adding unnecessary constraints, it may make a problem in some very niche case

Reply:
We did discuss whether we would need the INPC constraint and I believe we came to the conclusion that forcing this constraint would make people consider why they need it. I believe we also agreed that it wouldn't be a breaking change to remove the constraint rather than add it.